Allow nested scrolled element

zoltan-ky opened the pull request:
Hi,
Here are a few essential fixes to the javascript code that I ran across when working on my site with over 100 albums and 6000+ images.  The bugs were manifesting themselves as flickers on scroll and misaligned wide (panorama) images.  The diffs and inspection will reveal the problems in the code.

One small 'feature improvement' in `infinite_scroll.js`, that should be backward compatible, is to allow the scrolled (overflow) element to be nested inside the document and reached through `event.target`, something I needed due to the layout of our site.

(Should you care to see its operation, its our now 16-year old sailboatdazzle.com site (started under django 0.x) and presently under django 4.x and python 3.10...  Some time ago I was compelled to add an email-based registration start page with a simple captcha, no passwords required or generated, returns a link via email and remembers the session per device).

I hope this helps improve your app should you choose to pull this in.  Please let me know if...  I have further improvements in the python code, those I will have to put into another branch of my fork for review.

Thanks for the great app!  Best regards, Zoltan

Starcross commented:
Hi there, thanks for your PR and your kind words!

I did try checking out this but it appears to have stopped the justification working correctly and the infinite scroll is not triggering for some reason. This is using the base template, so I wonder if some changes you made to yours are needed?

Of course yes I am open to suggestions or other PRs to consider.

![Screenshot from 2023-01-04 21-28-47](https://user-images.githubusercontent.com/8641396/210653656-70204b1b-56f6-448b-98bc-39bda65120e8.png)

zoltan-ky commented:
Hi Alex,

I’ll look into it asap and get back …

Cheers,
Zoltan

> On Jan 4, 2023, at 4:31 PM, Alex ***@***.***> wrote:
> 
> 
> Hi there, thanks for your PR and your kind words!
> 
> I did try checking out this but it appears to have stopped the justification working correctly and the infinite scroll is not triggering for some reason. This is using the base template, so I wonder if some changes you made to yours are needed?
> 
> Of course yes I am open to suggestions or other PRs to consider.
> 
>  <https://user-images.githubusercontent.com/8641396/210653656-70204b1b-56f6-448b-98bc-39bda65120e8.png>
> —
> Reply to this email directly, view it on GitHub <https://github.com/Starcross/django-starcross-gallery/pull/17#issuecomment-1371448565>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ASQOAQW35M4T2RT53X6XG43WQXT3LANCNFSM6AAAAAATP7IEHE>.
> You are receiving this because you authored the thread.
>

zoltan-ky commented:
You were correct in me forgetting to submit a changed  templates/gallery/partials/thumbnail.html file - in it I had removed the `style="width=. {..}px"` attribute from <img>.  That made the previous commit work with your "New Zeland" example of 150 images.

However, I've since realized that these image-sizing-method (style-based vs intrinsic) changes are not necessary for infinite scrolling  so I reverted them.  This pull request thus becomes a small 'feature request' enhancement only.
